src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -223,6 +223,20 @@ internal void ClearAnalyzerExceptionHandlers(DiagnosticAnalyzer analyzer)
             }
         }
 
+        internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diagnostic, AnalyzerExecutor analyzerExecutor)
+        {
+            var supportedDescriptors = GetSupportedDiagnosticDescriptors(analyzer, analyzerExecutor);
+            foreach (var descriptor in supportedDescriptors)
```

Comment:
but we already have compiler analyzer that has many diagnostics for each compiler diagnostic reported we will linearly scan all those. and assumption seems a bit to risky. how about analyzer manager who cache the descriptor anyway, check the count of the descriptor when cache and if it is more than 10 or something, we cache them in hashset rather than array?